go/vt/vttablet/tabletserver/throttle/throttler.go

```diff
@@ -704,6 +715,11 @@ func (throttler *Throttler) generateTabletHTTPProbeFunction(ctx context.Context,
 		if checkResult.StatusCode == http.StatusInternalServerError {
 			mySQLThrottleMetric.Err = fmt.Errorf("Status code: %d", checkResult.StatusCode)
 		}
+		if checkResult.RecentlyChecked {
+			// We have just probed a tablet, and it reported back that someone just recently "check"ed it.
+			// We therefore renew the heartbeats lease.
+			go throttler.heartbeatWriter.RequestHeartbeats()
+		}
```

Comment:
My gut tells me there's the potential for a race here so that we don't request heartbeats for 1+ seconds which could throttle things and impact production traffic unnecessarily. Let's say that the throttler is checked once a second, but it's always just slightly behind the second tick.

Reply:
Ah, but heartbeats are leased for some `5s`-`10s`, and so there is no race. Leasing heartbeats to `1s` is unadvisable. Now that you mention it, I should probably enforce a reasonable lower limit.